packages/rollup-contracts/deploy/rollup-chain.ts

```diff
@@ -0,0 +1,105 @@
+/* External Imports */
+import { deploy, deployContract } from '@eth-optimism/core-utils'
+import { Wallet } from 'ethers'
+import { Provider } from 'ethers/providers'
+
+/* Internal Imports */
+import * as RollupMerkleUtils from '../build/RollupMerkleUtils.json'
+import * as CanonicalTransactionChain from '../build/CanonicalTransactionChain.json'
+import * as StateCommitmentChain from '../build/StateCommitmentChain.json'
+import * as SequencerBatchSubmitter from '../build/SequencerBatchSubmitter.json'
+import * as L1ToL2TransactionPasser from '../build/L1ToL2TransactionPasser.json'
+
+import { resolve } from 'path'
+
+const rollupChainDeploymentFunction = async (
+  wallet: Wallet,
+  provider: Provider
+): Promise<string> => {
+  const sequencer = process.env.SEQUENCER_PRIVATE_KEY
+    ? new Wallet(process.env.SEQUENCER_PRIVATE_KEY, provider)
+    : wallet
+  const inclusionPeriod = process.env.FORCE_INCLUSION_PERIOD || 600
+  const fraudVerifier = wallet // TODO actually deploy Fraud Verifier
+
+  console.log(`\nDeploying Rollup Chain!\n`)
+  console.log(`\nDeploying RollupMerkleUtils...\n`)
+  const rollupMerkleUtils = await deployContract(RollupMerkleUtils, wallet)
+
+  console.log(
+    `\nDeploying SequencerBatchSubmitter with Sequencer address: {sequencer.addres}...\n`
+  )
+  const sequencerBatchSubmitter = await deployContract(
+    SequencerBatchSubmitter,
+    wallet,
+    sequencer.address
+  )
+
+  console.log(`\nDeploying L1ToL2TransactionPasser...\n`)
+  const l1ToL2TransactionPasser = await deployContract(
+    L1ToL2TransactionPasser,
+    wallet
+  )
+
+  console.log(`\nDeploying CanonicalTransactionChain...\n`)
+  const canonicalTxChain = await deployContract(
+    CanonicalTransactionChain,
+    wallet,
+    rollupMerkleUtils.address,
+    sequencerBatchSubmitter.address,
+    l1ToL2TransactionPasser.address,
+    inclusionPeriod
+  )
+  const l1ToL2QueueAddress = await canonicalTxChain.l1ToL2Queue()
+  const safetyQueueAddress = await canonicalTxChain.safetyQueue()
+
+  console.log(`\nDeploying StateCommitmentChain...\n`)
+  const stateChain = await deployContract(
```

Comment:
Is it possible for any of this to already be deployed? If so, it might make sense to create individual functions for these that check if there's a configured address to use for a certain contract and only deploying if there is not one configured.

Reply:
So I feel like I can't think of any case where you wouldn't want to just deploy these all at once since they all depend on each other. I pulled out individual deployment functions and then realized it's such a pain to configure it to pass in the addresses that those deployments rely on.